tests/core/pyspec/eth2spec/test/sanity/test_blocks.py

```diff
@@ -256,11 +254,65 @@ def test_proposer_slashing(spec, state):
     assert get_balance(state, validator_index) < get_balance(pre_state, validator_index)
 
 
+@with_all_phases
+@spec_state_test
+def test_double_same_proposer_slashings_same_block(spec, state):
+    proposer_slashing = get_valid_proposer_slashing(spec, state, signed_1=True, signed_2=True)
+    validator_index = proposer_slashing.signed_header_1.message.proposer_index
+    assert not state.validators[validator_index].slashed
+
+    yield 'pre', state
+
+    block = build_empty_block_for_next_slot(spec, state)
+    block.body.proposer_slashings = [proposer_slashing, proposer_slashing]
+    signed_block = state_transition_and_sign_block(spec, state, block, expect_fail=True)
+
+    yield 'blocks', [signed_block]
+    yield 'post', None
+
+
+@with_all_phases
+@spec_state_test
+def test_double_similar_proposer_slashings_same_block(spec, state):
+    validator_index = spec.get_active_validator_indices(state, spec.get_current_epoch(state))[-1]
+
+    # Same validator, but different slashable offences in the same block
+    proposer_slashing_1 = get_valid_proposer_slashing(spec, state, random_root=b'\xaa' * 32,
+                                                      validator_index=validator_index,
+                                                      signed_1=True, signed_2=True)
+    proposer_slashing_2 = get_valid_proposer_slashing(spec, state, random_root=b'\xbb' * 32,
+                                                      validator_index=validator_index,
+                                                      signed_1=True, signed_2=True)
+    assert not state.validators[validator_index].slashed
+
+    yield 'pre', state
+
+    block = build_empty_block_for_next_slot(spec, state)
+    block.body.proposer_slashings = [proposer_slashing_1, proposer_slashing_2]
+    signed_block = state_transition_and_sign_block(spec, state, block, expect_fail=True)
+
+    yield 'blocks', [signed_block]
+    yield 'post', None
+
```

Comment:
Should we add `test_double_different_proposer_slashings_same_block`? to show that two proposer slashings _can_ happen in the different proposer case?